middleware.py

```diff
@@ -0,0 +1,33 @@
+"""
+Contains the MiddleWare Class for NodeBB
+"""
+
+import jwt
+
+from django.conf import settings as django_settings
+
+
+class UserSessionSharingMiddleware(object):
+    """
+    Middleware to set jwt login token on sign in
+    Used for session sharing with NodeBB community
+    """
+
+    def process_response(self, request, response):
+        try:
+            if request.user.is_authenticated():
+                user_data = {
+                    'id': request.user.id,
+                    'username': request.user.username,
+                    'email': request.user.email
+                }
+                jwt_secret = django_settings.OPENEDX_NODEBB_DISCUSSION['DISCUSSION_JWT_SECRET']
```

Comment:
`OPENEDX_NODEBB_DISCUSSION` is related to `endpoints, url, domain`
while `NODEBB_SETTINGS` is related to `authentication` and contains `tokens and keys`

Reply:
Okay, but why aren't we using a single key?

Reply:
Because in edX we keep our secrets in `auth.json` file and remaining settings in `env.json`. As we have 2 different dictionaries for both settings. We can keep settings in one variable while reading from this files but we have to make a nested dictionary for that.